### ISSUE TITLE
numpy.linalg.linalg.LinAlgError: SVD did not converge during rmgpy.reaction.Reaction.reverse_arrhenius_rate

### ISSUE THREAD

**Issue by rwest** [ReactionMechanismGenerator/RMG-Py#1833]
### Bug Description
During calls to `rmgpy.reaction.Reaction.reverse_arrhenius_rate` I am now quite often getting `numpy.linalg.linalg.LinAlgError: SVD did not converge in Linear Least Squares` errors, that cause RMG to immediately crash.

One context was @ajuluc and I working on a tool to check collision limit violations.
Another was our chemkin mechanism importer script. 

It might be that there are NAN values creeping in, but I don't see why that would happen in these contexts, and for the importer script it never used to be a problem, and it wasn't a problem when @chrisbNEU ran the same input on his ubuntu machine on Wednesday, IIRC. So I suspect it might be an OS or compiler or hardware specific problem.
Based on some discussion at https://github.com/numpy/numpy/issues/12941  etc.

Partly just sharing this here as a place for others to comment if they run into similar issues.

Here's some stack trace:
```
  File "rmgpy/reaction.py", line 791, in rmgpy.reaction.Reaction.generate_reverse_rate_coefficient
  File "rmgpy/reaction.py", line 872, in rmgpy.reaction.Reaction.generate_reverse_rate_coefficient
  File "rmgpy/reaction.py", line 854, in rmgpy.reaction.Reaction.generate_reverse_rate_coefficient
  File "rmgpy/reaction.py", line 829, in rmgpy.reaction.Reaction.generate_reverse_rate_coefficient
  File "rmgpy/reaction.py", line 788, in rmgpy.reaction.Reaction.reverse_arrhenius_rate
  File "rmgpy/kinetics/arrhenius.pyx", line 173, in rmgpy.kinetics.arrhenius.Arrhenius.fit_to_data
  File "/Users/rwest/anaconda/envs/rmg3/lib/python3.7/site-packages/numpy/linalg/linalg.py", line 2156, in lstsq
    x, resids, rank, s = gufunc(a, b, rcond, signature=signature, extobj=extobj)
  File "/Users/rwest/anaconda/envs/rmg3/lib/python3.7/site-packages/numpy/linalg/linalg.py", line 101, in _raise_linalgerror_lstsq
    raise LinAlgError("SVD did not converge in Linear Least Squares")
numpy.linalg.linalg.LinAlgError: SVD did not converge in Linear Least Squares
```

And here's my numpy config
```
$ python
Python 3.7.4 (default, Aug 13 2019, 15:17:50) 
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> np.show_config()
mkl_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/Users/rwest/anaconda/envs/rmg3/lib']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/Users/rwest/anaconda/envs/rmg3/include']
blas_mkl_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/Users/rwest/anaconda/envs/rmg3/lib']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/Users/rwest/anaconda/envs/rmg3/include']
blas_opt_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/Users/rwest/anaconda/envs/rmg3/lib']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/Users/rwest/anaconda/envs/rmg3/include']
lapack_mkl_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/Users/rwest/anaconda/envs/rmg3/lib']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/Users/rwest/anaconda/envs/rmg3/include']
lapack_opt_info:
    libraries = ['mkl_rt', 'pthread']
    library_dirs = ['/Users/rwest/anaconda/envs/rmg3/lib']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['/Users/rwest/anaconda/envs/rmg3/include']
```

### Installation Information
Describe your installation method and system information.
 - OS: Mac OS 10.14.6 
 - Installation method: from source in Anaconda environment
 - RMG version information:
   - RMG-Py: 2.4.1-866-g66b751b4c (on importer branch)

**Comment by rwest** [comment]
More relevant discussion here: 
https://github.com/numpy/numpy/issues/1588
https://github.com/ContinuumIO/anaconda-issues/issues/695
sounds like an issue with Intel's MKL library, which Continuum use to build the numpy distributed through anaconda. There are some hints at workarounds.

**Comment by rwest** [comment]
An example of some of Dave Agan's [Debugging Rules](http://debuggingrules.com/?page_id=40)!
I should have "_quit thinking and look_"ed a whole lot sooner. i.e. looked at the values I was trying to fit, rather than just googling the error message and finding what _other people's_ problems were.  Along the way I learned a lot about the differences between numpy and scipy's different linalg routines, and underlying Accelerate frameworks etc.

But that was all irrelevant. Turns out we were trying to fit `nan` values caused by taking logarithms of negative rates.

This is now fixed in pull request #1834
